pypeit/setup_gui/model.py

```diff
@@ -0,0 +1,1469 @@
+
+        return False
+
+    def flags(self, index):
+        """Returns item flags for a given index. Overridden method from QAbstractItemModel
+        
+        Args:
+            index (QModelIndex): The index to get flags for
+
+
+        Return:
+            flags (Qt.ItemFlag): The bitmask flags for the index. Currently only Qt.ItemFlag.ItemIsEditable
+                                 is supported.
+        """
+        base_flags = super().flags(index)
+        if self.colnames[index.column()] in self.editable_columns:
+            base_flags |= Qt.ItemFlag.ItemIsEditable
+        return base_flags
+
+    def headerData(self, section, orientation, role):
+        """Display header data for the table. For columns we give a column name, for rows we return
+        a number. Overridden method from QAbstractItemModel.
+
+        Args:
+            section (int): The "section" the header is used for. For table models this is the 
+                           zero based row or column number.
+            orientation (Qt.Orientation): Whether the horizontal (column header) or vertical (row header)
+                                          is being requested.
+            role (Qt.Role): The role header data is being requested for. This method supports
+                            Qt.InitialSortOrderRole and Qt.DisplayRole.
+
+        Returns:
+            str: The name for the column/row, or None if there is no applicable name for the given section/orientation/role.
+        """
+        if role == Qt.DisplayRole:
+            # Display the actual name of the header row/column
+            if orientation == Qt.Orientation.Horizontal and section < len(self.colnames):
+                # Columns have propper names
+                return self.colnames[section]
+            else:
+                return " "
+        else:
+            # A non-applicable role or a sort order request for a column that we're not sorted by.
+            return None
+
+    def getSetup(self, row : int) -> dict:
+        """Return the configuration/setup values for the given row of metadata.
+        
+        Args:
+            row: The row within the model to return the setup values for
+        Return:
+            A dictionary of configuration key/value pairs. An empty dictionary is returned
+            if there is no metadata or the spectrograph has no configuration keys
+        """
+        if self.metadata is None or row >= len(self.metadata):
+            return {}
+        return self.metadata.get_configuration(row)
+
+
+
+    def getDefaultColumns(self):
+        """Return the default columns to display to the user. This can vary based
+        on the current spectrograph.
+        
+        Return:
+        A list of column names, in order to display.
+        """
+        if self.metadata is not None:
+            default_columns = self.metadata.set_pypeit_cols(write_bkg_pairs=True)
+            return default_columns
+        else:
```

Comment:
Skip the middle man and don't need the else.  (Sorry for sounding like a broken record...)